### DISCUSSION TITLE
How to get the maximum overview level in a COG?

### DISCUSSION THREAD

**Discussion by rsignell** [corteva/rioxarray#756] [Q&A] [answered]
I'm trying to figure out how to find the maximum overview level in a COG, and I see the function here: 
https://rasterio.readthedocs.io/en/stable/api/rasterio.rio.overview.html#rasterio.rio.overview.get_maximum_overview_level
but I can't figure out how to use it.  

I thought this would work, but it doesn't:
``` python
import rioxarray 

https_url = 'https://s3.us-west-2.amazonaws.com/prod-is-usgs-sb-prod-publish/618e83cad34ec04fc9caa715/South_Carolina_CoNED_Topobathy_DEM_1m.tif'

da = rioxarray.open_rasterio(https_url)

da.rio.overview.get_maximum_overview_level()
```

**Comment by RichardScottOZ** [comment]
That is a rasterio function not rioxarray...so you would have to make rasterio available and use that I think?

**Comment by Kirill888** [comment]
It would be cool for rioxarray/rasterio to support numpy style of -ve indexing for overviews, with -1 meaning highest available overview level, that way one can avoid double open for some common use cases.

**Reply by snowman2** [reply]
It is handled by GDAL: https://gdal.org/api/raster_c_api.html#_CPPv410GDALOpenExPKcjPPCKcPPCKcPPCKc

If support for numpy style indexing were added there, it would work for rasterio & rioxarray.

**Comment by snowman2** [comment, marked as answer]
```python
import rasterio
import rioxarray

https_url = 'https://s3.us-west-2.amazonaws.com/prod-is-usgs-sb-prod-publish/618e83cad34ec04fc9caa715/South_Carolina_CoNED_Topobathy_DEM_1m.tif'

with rasterio.open(https_url) as rds:
    max_overview_level = len(rds.overviews(1)) - 1

da = rioxarray.open_rasterio(https_url, overview_level=max_overview_level)
```
